hipify-clang/packaging/hipify-clang.txt

```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(hipify-clang)
+
+install(PROGRAMS @HIPIFY_INSTALL_PATH@/hipify-clang DESTINATION bin)
+install(DIRECTORY @HIPIFY_INSTALL_PATH@/include DESTINATION bin)
+
+#############################
+# Packaging steps
+#############################
+set(CPACK_SET_DESTDIR TRUE)
+set(CPACK_INSTALL_PREFIX "/opt/rocm/hip")
```

Comment:
Do we have some variables for installing in a different location, which we should take into account here?

Reply:
Can you elaborate?
I had set that path just following the standard from hip_base.txt.

Reply:
> I had set that path just following the standard from hip_base.txt.
Yes, of course. I'm just wondering, do we have in HIP an opportunity to install not only by default path? Sometimes it is needed.

Reply:
> I'm just wondering, do we have in HIP an opportunity to install not only by default path? Sometimes it is needed.

In some contexts it is _required_.  For example `/opt` on the `summit` system is a network mounted filesystem and so they would like to be able to install things in a custom location, not the default.

Any packages created need that flexibility too; they should be relocatable to anywhere in the filesystem hierarchy.

Reply:
> 
> 
> > I'm just wondering, do we have in HIP an opportunity to install not only by default path? Sometimes it is needed.
> 
> In some contexts it is _required_. For example `/opt` on the `summit` system is a network mounted filesystem and so they would like to be able to install things in a custom location, not the default.
> 
> Any packages created need that flexibility too; they should be relocatable to anywhere in the filesystem hierarchy.

HIP supports CMAKE_INSTALL_PREFIX to install built HIP to a custom location, https://github.com/ROCm-Developer-Tools/HIP/blob/master/INSTALL.md#building-hip-from-source

Reply:
> HIP supports CMAKE_INSTALL_PREFIX to install built HIP to a custom location,

`CPACK_PACKAGE_VERSION` here is somewhat different subsequent story.

Reply:
So, should we be using the /opt/rocm/hip prefix or something else?

Reply:
> So, should we be using the /opt/rocm/hip prefix or something else?

I would say that we need something like `CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT` but for `CPACK_INSTALL_PREFIX`. Unfortunately, `CPACK_INSTALL_PREFIX` doesn't work like `CMAKE_INSTALL_PREFIX`:

```cmake
//No help, variable specified on the command line.
CPACK_INSTALL_PREFIX:UNINITIALIZED=dist
```